src/ui/Controls/NikseComboBox.cs

```diff
@@ -620,6 +565,74 @@ protected override void OnMouseDown(MouseEventArgs e)
             base.OnMouseDown(e);
         }
 
+        private void EnsureListViewInitialized()
+        {
+            if (_listView != null)
+            {
+                return;
+            }
+
+            _listView = new ListView();
+            _listView.View = View.Details;
+            _listView.Columns.Add("text", DropDownWidth - 4);
+            _listView.HeaderStyle = ColumnHeaderStyle.None;
+            _listView.FullRowSelect = true;
+            _listView.MultiSelect = false;
+            _listView.HideSelection = false;
+
+            if (Configuration.Settings.General.UseDarkTheme)
+            {
+                DarkTheme.SetDarkTheme(_listView);
+            }
+
+            _listView.MouseEnter += (sender, args) => { _hasItemsMouseOver = true; };
+            
+            _listView.KeyDown += (sender, args) =>
+            {
+                if (args.KeyCode == Keys.Escape)
+                {
+                    HideDropDown();
+                }
+                else if (args.KeyCode == Keys.Enter)
+                {
+                    _listViewMouseLeaveTimer.Stop();
+                    var item = _listView.SelectedItems[0];
+                    _selectedIndex = item.Index;
+                    _textBox.Text = item.Text;
+                    Invalidate();
+                    SelectedIndexChanged?.Invoke(this, EventArgs.Empty);
+                    HideDropDown();
+                    _textBox.Focus();
+                    _textBox.SelectionLength = 0;
+                }
+            };
+
+            _listView.MouseClick += (sender, mouseArgs) =>
+            {
+                if (mouseArgs == null)
+                {
+                    return;
+                }
+
+                var cachedCount = _listView.Items.Count;
```

Comment:
cache the count as `_listView.Items.Count` is very expensive!

See source code